surface_apps/iso_surfaces/utils.py

```diff
@@ -98,9 +100,16 @@ def interp_to_grid(  # pylint: disable=too-many-locals
     """
 
     grid = []
-    is_finite = np.isfinite(values)
-    locations = entity.locations[is_finite, :]
-    finite_values = values[is_finite]
+    is_finite = np.isfinite(data.values)
+    if isinstance(entity, Octree):
```

Comment:
I would generalize it to `hasattr(entity, "centroids")`

Reply:
done.